### PR TITLE
Added `<FxDropdown />`

### PR DESCRIPTION
**Pull request by buchansm**
Affiliated with: #168 

drop-down

### CODE REVIEW

**Comment by JDMathew** [comment]
Is this `feat/form-picker` branched off `task/form-demo-screen`? Might have some conflicts with duplicated commits when merging back into `main` if that is the case.

**Comment by buchansm** [comment]
Yeah it is. I noticed that as well 😞.

**Comment by buchansm** [comment]
I just fixed the conflicts @JDMathew.

File: libs/component-library/src/lib/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  const dropdownLabel =
+    options.find(({ value }) => value === selectedValue)?.label ??
+    options[0].label;
+
+  const onPress = () => {
+    bottomSheetRef.current?.present();
+    setFocus(true);
+  };
+
+  const onDismiss = () => setFocus(false);
```
Comment (JDMathew) [inline libs/component-library/src/lib/dropdown/dropdown.tsx:82]: Potentially give the consumer of the dropdown the onDismiss prop callback. i.e:

```typescript
const _onDismiss = () => {
  setFocus(false);
  onDismiss();
};
```
.
.
```jsx
<FxBottomSheetModal
  ref={bottomSheetRef}
  title={title}
  onDismiss={_onDismiss}
>
```

**Review by JDMathew — APPROVED** [review]
LGTM. Feel free to make suggested changes

File: libs/component-library/src/lib/dropdown/dropdown.tsx (hunk 1 above)
Comment (buchansm) [reply libs/component-library/src/lib/dropdown/dropdown.tsx:82]: Nice, yeah I can do that.

**Comment by buchansm** [comment]
I had considered auto closing, but felt like it could turn into a point of frustration for the user when traversing longer option lists so I opted to not implement it.

**Pull request merged**